Add back lost special deserialization behavior

kangalio opened the pull request:
https://discord.com/channels/381880193251409931/381912587505500160/1043283742254448820

> i forgot to port some of the derive-deviating behavior to the new Deserialize impls in https://github.com/serenity-rs/serenity/pull/2287
> namely, the fact that `user` is actually not a required field and is provided by serenity in guilds by cloning `member.user`

kangalio committed:
Add back lost special deserialization behavior

mkrasnitski commented on src/model/application/interaction/application_command.rs:
```diff
@@ -242,6 +242,11 @@ impl<'de> Deserialize<'de> for CommandInteraction {
+                // If `member` is present, `user` wasn't sent and is still filled with default data
+                interaction.user = member.user.clone();
+            }
             interaction.data.resolved.roles.values_mut().for_each(|r| r.guild_id = guild_id);
```
After taking a closer look at #2287, I noticed this line was missing from the implementations for `ComponentInteraction` and `ModelInteraction`, even though they both previously called `add_guild_id_to_resolved`. Was this intentional on your part?

kangalio replied:
Yes, that was actually intentional. add_guild_id_to_resolved did two things:
- insert guild_id into Member
- insert guild_id into CommandData::resolved::roles

The latter only applies to CommandInteraction's so I just inlined the former into ComponentInteraction and ModalInteraction.

kangalio committed:
Add #[serde(default)] to user fields

To fix missing fields error. The defaulted value will be filled in by member.user

kangalio committed:
Forgot to commit this

kangalio commented:
Can this PR be merged? I've now run into the bug myself while testing the `/modal` command of my test bot :sweat_smile: 
![Screenshot_20221120_171421](https://user-images.githubusercontent.com/21220820/202913075-e1c662c1-01db-4b5c-b657-874b5eacbb90.png)
